portal-impl/src/com/liferay/portal/jsonwebservice/JSONWebServiceActionParameters.java

```diff
@@ -214,10 +221,43 @@ private void _collectFromRequestParameters(HttpServletRequest request) {
 		for (String parameterName : parameterNames) {
 			Object value = null;
 
-			if ((uploadServletRequest != null) &&
-				(uploadServletRequest.getFileName(parameterName) != null)) {
+			if ((multipartParameterMap != null) &&
+				multipartParameterMap.containsKey(parameterName)) {
+
+				FileItem[] fileItems = multipartParameterMap.get(parameterName);
+
+				File[] files = new File[fileItems.length];
+
+				for (int i = 0; i < fileItems.length; i++) {
+					FileItem fileItem = fileItems[i];
+
+					File file = fileItem.getStoreLocation();
 
-				value = uploadServletRequest.getFile(parameterName, true);
+					if (fileItem.isInMemory()) {
+						try {
+							FileUtil.write(file, fileItem.getInputStream());
+
+							file.deleteOnExit();
```

Comment:
hey @topolik, 
I am a bit concerned about this... are these files deleted elsewhere? I am weary that the JVM is not expected to be closed so often and this could fill the disks?